detoks/src/main/java/nl/tudelft/trustchain/detoks/Strategy.kt

```diff
@@ -51,17 +104,38 @@ private val strategyComparators = mutableMapOf<Int, (Pair<TorrentHandler, Profil
         handlers: MutableList<TorrentHandler>,
         profiles: HashMap<String, ProfileEntry>
     ): MutableList<TorrentHandler> {
-        if (id == 0) return handlers.shuffled().toMutableList()
+        if (id == STRATEGY_RANDOM) return handlers.shuffled().toMutableList()
         if (!strategyComparators.contains(id)) return handlers
 
-        val handlerProfile = handlers.map {
+        var handlerProfile = handlers.map {
             val key = it.handle.infoHash().toString()
             if (!profiles.contains(key)) return@map Pair(it, ProfileEntry())
             return@map Pair(it, profiles[key])
         }
-        val sortedHandlerProfile =
+
+        var sortedHandlerProfile =
             handlerProfile.sortedWith(strategyComparators[id]!!).toMutableList()
 
+        if (id == STRATEGY_RISING){
+            val filteredLow = handlerProfile.filter{ cutoffComparator(it, RISING_CUTOFF_SECONDS) }
+            val filteredHigh = handlerProfile.filterNot{ cutoffComparator(it, RISING_CUTOFF_SECONDS) }
+
+            val sortedLow = filteredLow.sortedWith(strategyComparators[id]!!).toMutableList()
+            val sortedHigh = filteredHigh.sortedWith(strategyComparators[id]!!).toMutableList()
+
+            sortedHandlerProfile = (sortedLow + sortedHigh).toMutableList()
+        }
+
+        if (id == STRATEGY_HOT) {
+            val filteredLow = handlerProfile.filter{ cutoffComparator(it, HOT_CUTOFF_SECONDS) }
+            val filteredHigh = handlerProfile.filterNot{ cutoffComparator(it, HOT_CUTOFF_SECONDS) }
+
+            val sortedLow = filteredLow.sortedWith(strategyComparators[id]!!).toMutableList()
+            val sortedHigh = filteredHigh.sortedWith(strategyComparators[id]!!).toMutableList()
+
+            sortedHandlerProfile = (sortedLow + sortedHigh).toMutableList()
```

Comment:
I think you can still combine more of this into a separate method, since there is still a lot double, and only diff is the cutoff constant. Also, in line 116 you already sort, and in these if's you do it again, so essentially doing double work. Maybe just add a switch, if it is a strategy that requires a cut-off call one method who's returned value is assigned to `sortedHandlerProfile`, otherwise do the sorting right away and assign to `sortedHandlerProfile`